fish-rust/src/builtins/ulimit.rs

```diff
@@ -0,0 +1,354 @@
+        multiplier: 1024,
+    },
+    #[cfg(any(target_os = "android", target_os = "linux"))]
+    Resource {
+        resource: resource::Resource::RLIMIT_NICE,
+        desc: L!("Control of maximum nice priority"),
+        switch_char: 'e',
+        multiplier: 1,
+    },
+    Resource {
+        resource: resource::Resource::RLIMIT_FSIZE,
+        desc: L!("Maximum size of files created by the shell"),
+        switch_char: 'f',
+        multiplier: 1024,
+    },
+    #[cfg(any(target_os = "android", target_os = "linux"))]
+    Resource {
+        resource: resource::Resource::RLIMIT_SIGPENDING,
+        desc: L!("Maximum number of pending signals"),
+        switch_char: 'i',
+        multiplier: 1,
+    },
+    #[cfg(any(
+        target_os = "android",
+        target_os = "freebsd",
+        target_os = "openbsd",
+        target_os = "linux",
+        target_os = "netbsd"
+    ))]
+    Resource {
+        resource: resource::Resource::RLIMIT_MEMLOCK,
+        desc: L!("Maximum size that may be locked into memory"),
+        switch_char: 'l',
+        multiplier: 1024,
+    },
+    #[cfg(any(
+        target_os = "android",
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "openbsd",
+        target_os = "linux",
+    ))]
+    Resource {
+        resource: resource::Resource::RLIMIT_RSS,
+        desc: L!("Maximum resident set size"),
+        switch_char: 'm',
+        multiplier: 1024,
+    },
+    Resource {
+        resource: resource::Resource::RLIMIT_NOFILE,
+        desc: L!("Maximum number of open file descriptors"),
+        switch_char: 'n',
+        multiplier: 1,
+    },
+    #[cfg(any(target_os = "android", target_os = "linux"))]
+    Resource {
+        resource: resource::Resource::RLIMIT_MSGQUEUE,
+        desc: L!("Maximum bytes in POSIX message queues"),
+        switch_char: 'q',
+        multiplier: 1024,
+    },
+    #[cfg(any(target_os = "android", target_os = "linux"))]
+    Resource {
+        resource: resource::Resource::RLIMIT_RTPRIO,
+        desc: L!("Maximum realtime scheduling priority"),
+        switch_char: 'r',
+        multiplier: 1,
+    },
+    Resource {
+        resource: resource::Resource::RLIMIT_STACK,
+        desc: L!("Maximum stack size"),
+        switch_char: 's',
+        multiplier: 1024,
+    },
+    Resource {
+        resource: resource::Resource::RLIMIT_CPU,
+        desc: L!("Maximum amount of CPU time in seconds"),
+        switch_char: 't',
+        multiplier: 1,
+    },
+    #[cfg(any(
+        target_os = "android",
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "openbsd",
+        target_os = "linux",
+    ))]
+    Resource {
+        resource: resource::Resource::RLIMIT_NPROC,
+        desc: L!("Maximum number of processes available to current user"),
+        switch_char: 'u',
+        multiplier: 1,
+    },
+    #[cfg(not(any(target_os = "freebsd", target_os = "netbsd", target_os = "openbsd")))]
```

Comment:
This should probably not be `not(bsd)` but rather the supported platforms. 

RLIMIT_AS is available on macOS but it's broken and not to be used. Same with RLIMIT_DATA.
Much has been written about macOS w/ RLIMIT_DATA and RLIMIT_AS:

https://bugs.chromium.org/p/chromium/issues/detail?id=853873#c2
https://stackoverflow.com/questions/15633474/python-on-macos-totally-ignoring-rlimit
https://rdrr.io/cran/unix/man/rlimit.html
https://github.com/avast-tl/retdec/issues/379

This would be a change from fish 3.6.0 so maybe it should be held back for a later release (cc @ridiculousfish)